clinica/iotools/utils/data_handling/_centering.py

```diff
@@ -247,24 +265,42 @@ def _compute_l2_norm(file_couples: List[Tuple[Path, Path]]) -> List[float]:
 
 def _find_files_with_modality(
     files_dir: Path, modalities: Optional[Iterable[str]] = None
-) -> List[Path]:
+) -> list[Path]:
     nifti_files = files_dir.rglob("*.nii*")
     nifti_files_filtered = []
 
     if modalities is None:
-        return nifti_files
+        return list(nifti_files)
     for f in nifti_files:
         if any(modality.lower() in f.name.lower() for modality in modalities):
             nifti_files_filtered.append(f)
     return nifti_files_filtered
 
 
+def _validate_bids_and_output_dir(
+    bids_dir: PathLike, output_dir: PathLike
+) -> Tuple[Path, Path]:
```

Comment:
Same as for List vs. list:

```suggestion
) -> tuple[Path, Path]:
```